R/source.R

```diff
@@ -435,29 +435,28 @@ check_cargo_output <- function(compilation_result, message_buffer, tty_has_color
   )
 
   if (!isTRUE(compilation_result$status == 0)) {
-    error_messages <- gather_cargo_output(
-      cargo_output,
-      "error",
-      tty_has_colors
-    )
-
-    errors_formatted <- purrr::map_chr(
-      error_messages,
-      cli::format_inline,
-      keep_whitespace = TRUE
-    ) %>%
-      # removing double new lines with single new line
-      stringi::stri_replace_all_fixed("\n\n", "\n") %>%
-      # ensures that the leading cli style `x` is there
-      rlang::set_names("x")
-
-    rlang::abort(
-      c(
-        "Rust code could not be compiled successfully. Aborting.",
-        unlist(errors_formatted)
-      ),
-      call = call,
-      class = "rextendr_error"
+    error_messages <-
+      gather_cargo_output(
+        cargo_output,
+        "error",
+        tty_has_colors
+      ) %>%
+        purrr::map_chr(
+          cli::format_inline,
+          keep_whitespace = TRUE
+        ) %>%
+        # removing double new lines with single new line
+        stringi::stri_replace_all_fixed("\n\n", "\n") %>%
+        # ensures that the leading cli style `x` is there
+        rlang::set_names("x")
+
+      rlang::abort(
+        c(
+          "Rust code could not be compiled successfully. Aborting.",
+          error_messages
+        ),
+        call = call,
```

Comment:
```suggestion
        stringi::stri_replace_all_fixed("\n\n", "\n")

      rlang::abort(
        c(
          "Rust code could not be compiled successfully. Aborting.",
          x = error_messages
        ),
        call = call,
```

Reply:
I don't think this is desired. 

See the below. `rlang::abort()` does not preprend the little `x` that are currently there for each error message from cargo. 

<img width="590" alt="image" src="https://user-images.githubusercontent.com/13534508/233982509-bd084618-4b8c-4635-bc21-4ac482ee6944.png">

Compare that with the existing formatting that this code keeps: 
<img width="570" alt="image" src="https://user-images.githubusercontent.com/13534508/233982706-92383e72-0dfe-4a3c-8e20-c56ccde177d5.png">

And compared with using `cli::cli_abort()`
<img width="713" alt="image" src="https://user-images.githubusercontent.com/13534508/233982828-cf7e7d2f-eb0d-4203-b7e7-07807cbde412.png">

Reply:
Your comment seems to be about something that my suggestion doesn't impact. But still, if this is the case, go for `cli::cli_abort`. I must have misunderstood something.

Reply:
Please excuse the screenshots. Reprex does not render these characters properly for me. 
Here is a reprex that you can run that might visualize the differences a bit better.

Essentially, in order to keep cargo formatting and to have the cli style `x` preprended to each error message, each element of the `error_message` vector must have the name `x` hence the `rlang::set_names()` call. Using `cli` will lose all formatting without the messy `cli::cli_fmt()` function that I previously had. Setting `x = error_message` inside of `rlang::abort()` doesn't do anything in this case because `error_messages` is likely > length 1. 

This is the cleanest iteration of cargo error formatting that I could devise. 

```r
first_error <-  "Rust code could not be compiled successfully. Aborting."

# truncated some error messages.
error_messages <- c("\033[1m\033[38;5;9merror\033[39m: expected one of `:`, `@`, or `|`, found `)`\033[22m\n \033[1m\033[38;5;12m--> \033[39m\033[", "\033[1m\033[38;5;9merror\033[39m: expected `:`\033[22m\n \033[1m\033[38;5;12m--> \033[39m\033[22msrc/lib.rs:3:13\n  \033[1m\033[38;5", "\033[1m\033[38;5;9merror[E0425]\033[39m: cannot find function `meta__hello` in this scope\033[22m\n \033[1m\033[38;5;12", "\033[1m\033[38;5;9merror\033[39m: aborting due to 3 previous errors\033[22m\n")

# rlang keeps any formatting that has been provided
# note that each element of `error_messages` gets a bullet
rlang::abort(
  c(first_error, error_messages),
)


# if we set use_cli_format = TRUE and name the second vector val 
# to `x = error_messages` we don't get the nice `x` in front of the
# error as we'd like _and_ we lose formatting provided by cargo
rlang::abort(
  c(first_error, x = error_messages),
  use_cli_format = TRUE
)

# if, however, we name all of the elements `x` and pass it
# directly into `rlang::abort()` we get the formatting we desire
rlang::abort(
  c(first_error, rlang::set_names(error_messages, "x"))
)
```
<img width="537" alt="image" src="https://user-images.githubusercontent.com/13534508/234004504-4c14dba0-b4e4-4c04-a53a-d0a57a9fd96b.png">